docs/guides/filesystem-usage.md

```diff
@@ -0,0 +1,186 @@
+# How to use the lakeFS file system
+
+This guide contains instructions and code snippets on how to use the lakeFS file system.
+
+## The lakeFS URI structure
+
+In the following subsections, we frequently make use of [lakeFS URIs](https://docs.lakefs.io/understand/model.html#lakefs-protocol-uris) in the example code.
+lakeFS URIs identify resources in a lakeFS deployment through a unique path consisting of repository name, lakeFS revision/ref name, and file name relative to the repository root.
+
+As an example, a URI like `repo/main/file.txt` addresses the `file.txt` file on the `main` branch in the repository named `repo`.
+
+In some lakeFS file system operations, directories are also allowed as resource names.
+For example, the URI `repo/main/data/` (note the trailing slash) refers to the `data` directory on the `main` branch in the `repo` repository.
+
+## On staged versus committed changes
+
+When uploading, copying, or removing files or directories from a branch, those removal operations are staged changes until another commit is created.
+`lakefs-spec` does not create these commits automatically, since it separates file operations from versioning operations rigorously.
+If you want to conduct versioning operations, like creating commits, between file transfers, the best way to do so is by using a **filesystem transaction**.
```

Comment:
Start a new paragraph here for readability.

I would go ahead and create a `TODO` placeholder page in the user guide for the transactions, so we can already link it here.

Reply:
Done in [8cb6e79](https://github.com/aai-institute/lakefs-spec/pull/156/commits/8cb6e79eb899e72e761a99bcc1827c728e808cfb).